charts/trino/templates/tests/test-graceful-shutdown.yaml

```diff
@@ -0,0 +1,51 @@
+{{- if .Values.worker.gracefulShutdown.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "trino.fullname" . }}-workers
+  labels:
+    {{- include "trino.labels" . | nindent 4 }}
+    app.kubernetes.io/component: test
+    test: graceful-shutdown
+  annotations:
+    "helm.sh/hook": test
+    "helm.sh/hook-weight": "1"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  type: ClusterIP
+  ports:
+    - port: {{ .Values.service.port }}
+      targetPort: http
+      protocol: TCP
+      name: http
+  selector:
+    {{- include "trino.selectorLabels" . | nindent 4 }}
+    app.kubernetes.io/component: worker
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: {{ include "trino.fullname" . }}-test-graceful-shutdown
+  labels:
+    {{- include "trino.labels" . | nindent 4 }}
+    app.kubernetes.io/component: test
+    test: graceful-shutdown
+  annotations:
+    "helm.sh/hook": test
+    "helm.sh/hook-weight": "2"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  containers:
+    - name: graceful-shutdown
+      image: curlimages/curl:latest
+      command: ["sh", "-c"]
+      args:
+        - >-
+          curl -v -X PUT
```

Comment:
Shouldn't this kill the worker pods, to verify the `preStop` lifecycle hook? I also don't see any assertion - ideally this should start a long-running query (a select query on the blackhole connector with `page_processing_delay` set) and verify it was able to finish before the cluster was restarted - verify the uptime.

Reply:
The ` --fail-with-body` will return non-zero exit code if the `curl` call returns any HTTP code >= 400.
So basically I am testing that the graceful shutdown authorization was configured correctly.
But I get your point, I think I can write a better test, maybe using a kubectl image and actually killing the worker pod.